### PR TITLE
services

### PR DESCRIPTION
**Pull request by marcelmoosbrugger**
__tasks:__

- [ ] feedback
- [x] mediator service
- [x] util service

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | None
| BC breaks        | None
| Documentation PR | None

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -15,3 +15,6 @@
         - 'changeButton' new instead of 'type: "select"'
         - 'translate' removed
         - 'languageNamespace' removed
+
+### Util-Extension
+- events 'husky.util.load.data' and 'husky.util.save.data' got removed
```
Comment (wachterjohannes) [inline UPGRADE.md]: please write this also in the SULU upgrade

File: husky_services/mediator.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * with this source code in the file LICENSE.
+ */
+
+define(function() {
+
+    'use strict';
+
+    var instance = null;
+
+    function Mediator() {}
```
Comment (wachterjohannes) [inline husky_services/mediator.js]: missing `;`

Comment (wachterjohannes) [inline husky_services/mediator.js]: `var Mediator = function(){};`

File: husky_services/mediator.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    Mediator.getInstance = function() {
+        if (instance == null) {
+            instance = new Mediator();
+        }
+        return instance;
+    };
+
+    Mediator.prototype.on = window.Husky.on;
```
Comment (wachterjohannes) [inline husky_services/mediator.js]: add JSDoc

File: husky_services/util.js [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+        entityMap = {
+            "&": "&amp;",
+            "<": "&lt;",
+            ">": "&gt;",
+            '"': '&quot;',
+            "'": '&#39;',
+            "/": '&#x2F;'
+        };
+
+    function Util() {
```
Comment (wachterjohannes) [inline husky_services/util.js]: `var util = function(){};`

File: husky_services/util.js [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+        if (s === 'object') {
+            if (value) {
+                if (value instanceof Array) {
+                    s = 'array';
+                }
+            } else {
+                s = 'null';
+            }
+        }
+        return s;
```
Comment (wachterjohannes) [inline husky_services/util.js]: empty line before return

File: husky_services/util.js [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+    };
+
+    Util.prototype.isEqual = _.isEqual;
```
Comment (wachterjohannes) [inline husky_services/util.js]: add basic JSDOC to all services methods

File: husky_services/mediator.js (hunk 2 above)
Comment (marcelmoosbrugger) [inline husky_services/mediator.js]: ; is not common with this syntax. Moreover in my opinion the syntax "function Mediator()" makes more explicit that it gets used as a constructor

**Pull request merged**